Drop explicit -fomit-frame-pointer instructions

"-fomit-frame-pointer" is part of the amd-64 default calling conventions, so
there is no use in explicitly specifying it on that platform.

The flag might also create larger (although faster) executables, so it might
be undesired to use on installations with a rather limited amount of memory.

It can even be counter-productive to care about the frame pointer omission
setting in amd64, such as when x86-32 versions use the inverted flag
"-fno-omit-frame-pointer" is used for debugging code, because in amd64 this is
*not* what would be expected.

This patch therefore drops any explict frame pointer handling options from the
unpatched original 32-bit versions of the template files and reverts to using
the compiler's default setting instead.

Index: env/set-heavily-optimized-gcc3.tpl
===================================================================
--- env.orig/set-heavily-optimized-gcc3.tpl
+++ env/set-heavily-optimized-gcc3.tpl
@@ -1,2 +1,2 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fomit-frame-pointer -fno-stack-check";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-stack-check";;
 CPPFLAGS) SET="-DNDEBUG";;
Index: env/set-heavily-optimized.tpl
===================================================================
--- env.orig/set-heavily-optimized.tpl
+++ env/set-heavily-optimized.tpl
@@ -1,2 +1,2 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fomit-frame-pointer -fno-stack-check";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-stack-check";;
 CPPFLAGS) SET="-DNDEBUG";;
Index: env/set-not-optimized.tpl
===================================================================
--- env.orig/set-not-optimized.tpl
+++ env/set-not-optimized.tpl
@@ -1,2 +1,2 @@
-CFLAGS | CXXFLAGS) SET="-O0 -fomit-frame-pointer -fno-stack-check";;
+CFLAGS | CXXFLAGS) SET="-O0 -fno-stack-check";;
 CPPFLAGS) SET="-DNDEBUG";;
Index: env/set-splitdebug-optimized.tpl
===================================================================
--- env.orig/set-splitdebug-optimized.tpl
+++ env/set-splitdebug-optimized.tpl
@@ -1,3 +1,3 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fomit-frame-pointer -fno-stack-check -g -feliminate-unused-debug-symbols";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-stack-check -g -feliminate-unused-debug-symbols";;
 CPPFLAGS) SET="-DNDEBUG";;
 FEATURES) SET="splitdebug";;
Index: env/set-conservative-srcdebug-optimized.tpl
===================================================================
--- env.orig/set-conservative-srcdebug-optimized.tpl
+++ env/set-conservative-srcdebug-optimized.tpl
@@ -1,3 +1,3 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fomit-frame-pointer -fno-stack-check -g -feliminate-unused-debug-symbols";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-stack-check -g -feliminate-unused-debug-symbols";;
 CPPFLAGS) SET="-DNDEBUG";;
 FEATURES) SET="installsources splitdebug";;
Index: env/set-conservatively-optimized.tpl
===================================================================
--- env.orig/set-conservatively-optimized.tpl
+++ env/set-conservatively-optimized.tpl
@@ -1,2 +1,2 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fno-strict-aliasing -fomit-frame-pointer -fno-stack-check";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-strict-aliasing -fno-stack-check";;
 CPPFLAGS) SET="-DNDEBUG";;
Index: env/set-split-fulldebug-optimized.tpl
===================================================================
--- env.orig/set-split-fulldebug-optimized.tpl
+++ env/set-split-fulldebug-optimized.tpl
@@ -1,3 +1,3 @@
-CFLAGS | CXXFLAGS) SET="-O3 -fomit-frame-pointer -fno-stack-check -ggdb3 -feliminate-unused-debug-symbols";;
+CFLAGS | CXXFLAGS) SET="-O3 -fno-stack-check -ggdb3 -feliminate-unused-debug-symbols";;
 CPPFLAGS) SET="-DNDEBUG";;
 FEATURES) SET="installsources splitdebug";;
